src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java

```diff
@@ -6,34 +6,33 @@
 import uk.ac.ebi.eva.submission.model.SubmissionStatus;
 import uk.ac.ebi.eva.submission.repository.SubmissionRepository;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.nio.file.attribute.PosixFilePermission;
 import java.time.LocalDateTime;
-import java.util.HashSet;
-import java.util.Set;
 import java.util.UUID;
 
 @Service
 public class SubmissionService {
 
-    private SubmissionRepository submissionRepository;
+    private final SubmissionRepository submissionRepository;
+    private final GlobusDirectoryProvisioner globusDirectoryProvisioner;
 
-    @Value("${eva.submission.dropbox}")
-    private String submissionDropbox;
+    @Value("${globus.uploadHttpDomain}")
+    private String uploadHttpDomain;
 
-    public SubmissionService(SubmissionRepository submissionRepository) {
+    public SubmissionService(SubmissionRepository submissionRepository,
+                             GlobusDirectoryProvisioner globusDirectoryProvisioner) {
         this.submissionRepository = submissionRepository;
+        this.globusDirectoryProvisioner = globusDirectoryProvisioner;
     }
 
-    public Submission initiateSubmission() {
-        String submissionId = createSubmissionDirectory();
+    public Submission initiateSubmission(String userId) {
+        String submissionId = UUID.randomUUID().toString();
+        String directoryToCreate = String.format("%s/%s", userId, submissionId);
+        globusDirectoryProvisioner.createSubmissionDirectory(directoryToCreate);
 
         Submission submission = new Submission(submissionId);
         submission.setStatus(SubmissionStatus.OPEN.toString());
         submission.setInitiationTime(LocalDateTime.now());
+        submission.setUploadUrl(uploadHttpDomain + "/" + directoryToCreate);
```

Comment:
I think we should also persist the userId

Reply:
Addressed